pkg/api/meta/interfaces.go

```diff
@@ -29,37 +29,48 @@ type VersionInterfaces struct {
 	MetadataAccessor
 }
 
-// Interface lets you work with object and list metadata from any of the versioned or
+type ObjectMetaAccessor interface {
+	GetObjectMeta() Object
+}
+
+// Object lets you work with object metadata from any of the versioned or
 // internal API objects. Attempting to set or retrieve a field on an object that does
 // not support that field (Name, UID, Namespace on lists) will be a no-op and return
 // a default value.
-// TODO: rename to ObjectInterface when we clear up these interfaces.
-type Interface interface {
-	TypeInterface
-
-	Namespace() string
+type Object interface {
+	GetNamespace() string
 	SetNamespace(namespace string)
-	Name() string
+	GetName() string
```

Comment:
Yeah, this is a problem with value types in go.  If you want value
accessors, you have to name them differently (we have Name defined, we need
to change to GetName).  Such is life.

On Fri, Dec 11, 2015 at 8:05 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/meta/interfaces.go
> <https://github.com/kubernetes/kubernetes/pull/18473#discussion_r47352598>
> :
>
> >  	SetNamespace(namespace string)
> > -	Name() string
> > +	GetName() string
>
> Yay Java? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18473/files#r47352598>.
>